torch/distributed/_tensor/placement_types.py

```diff
@@ -270,6 +293,17 @@ def __post_init__(self) -> None:
         if self.ndim == -1:
             self.ndim = len(self.shape)
 
+    def __hash__(self) -> int:
+        return hash((self.mesh, tuple(self.placements), self.shape))
```

Comment:
Do we also want to include "ndim"?

Reply:
probably no, `ndim` is inferred from shape, so it represents the same thing and should hash to the same. Also we might want to get rid of ndim and use shape uniformly.